services/bettersocial/post/blockAnonymousComment.js

```diff
@@ -0,0 +1,116 @@
+const UserBlockUserFunction = require('../../../databases/functions/userBlockUser');
+const UserFollowUserFunction = require('../../../databases/functions/userFollowUser');
+const UsersFunction = require('../../../databases/functions/users');
+const {
+  User,
+  UserFollowUser,
+  UserFollowUserHistory,
+  UserBlockedUser,
+  UserBlockedUserHistory,
+  sequelize
+} = require('../../../databases/models');
+const Getstream = require('../../../vendor/getstream');
+const RedisBlockHelper = require('../../redis/helper/RedisBlockHelper');
+const BetterSocialScoreBlockUser = require('../score/block-user');
+
+/**
+ * @typedef {Object} BetterSocialBlockAnonymousPostV2OptionalParams
+ * @property {Object[]} [reason]
+ * @property {string} [message]
+ */
+/**
+ *
+ * @param {string} token
+ * @param {string} selfUserId
+ * @param {string} postId
+ * @param {string} source
+ * @param {BetterSocialBlockAnonymousPostV2OptionalParams} params
+ * @returns
+ */
+const BetterSocialBlockAnonymousComment = async (
+  res,
+  token,
+  selfUserId,
+  postId,
+  commentId,
+  source,
+  params = {}
+) => {
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The function `BetterSocialBlockAnonymousComment` has a lot of parameters which can make it hard to understand and maintain. Consider using an object as a parameter to encapsulate these values. This will also make the function call more readable, as it will be clear what each value is for.

```diff
- const BetterSocialBlockAnonymousComment = async (
-   res,
-   token,
-   selfUserId,
-   postId,
-   commentId,
-   source,
-   params = {}
- ) => {
+ const BetterSocialBlockAnonymousComment = async ({
+   res,
+   token,
+   selfUserId,
+   postId,
+   commentId,
+   source,
+   params = {}
+ }) => {
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->